src/toil/job.py

```diff
@@ -3026,3 +3031,22 @@ def convertPromises(kwargs):
                 foundPromisedRequirement = True
         return foundPromisedRequirement
 
+
+class JobStoreFile(str):
+    """
+    A class to wrap the job store file id returned by writeGlobalFile and any attributes we may want
+    to add to it.
+    """
+    # Need to modify this so this subclass of the mutable str can accept two arguments.
+    @classmethod
+    def __new__(cls, *args):
+        # args is (<class>, fileStoreID, filePath) during instantiation, and (<class>, fileStoreID)
+        # during unpickling.  We care about the fileStoreID.
+        return str.__new__(cls, args[1])
```

Comment:
The @classmethod/@staticmethod decorator is unnecessary and I prefer calling out the initial argument to super's __new__. Your use of mutable makes no sense in this context. `str` instances aren't mutable.